README.md

```diff
@@ -142,11 +118,82 @@ pytest --runslow
+> If you have the repository cloned, and one of the conda environments described below active,
+> the following steps will build the documentation into `docs/_build/html`:
+>
+>     # Installation of the required Python packages to build the documentation only needs to
+>     # be done once per environment. Make sure you are in an active environment before doing
+>     # this installation.
+>     conda install -c conda-forge sphinx sphinx-autodoc-typehints sphinx_rtd_theme
+>
+>     cd docs/
+>     make html
+
+There are examples in the [examples folder](https://github.com/LSSTDESC/firecrown/examples)
+that show some use of Firecrown with both Cobaya and CosmoSIS.
+
+## Contributing
+
+Contributions to Firecrown are welcome.
+
+For any contribution, please start by [opening an issue](https://github.com/LSSTDESC/firecrown/issues),
+and using the GitHub interface to create a branch for that issue.
+
+When you have completed the task, push your commits to the branch you created for the issue and create a pull request.
+
+We are using several tools to help keep the code tidy and correct; these are described below.
+
+### Type checking
+
+We are using type-hinting in (most of) the code, to help ensure correct use of the framework.
+We are using `mypy` to verify the code is conforming to these type hints.
+Please run:
+
+```bash
+mypy firecrown/ --ignore-missing-imports 
+```
+
+and fix any errors reported before pushing commits to the GitHub repository.
+
+### Testing
+
+> Warning
+>
+> We are working on improving the coverage of testing for Firecrown; it is currently very inadequate.
+> As the coverage improves, we will provide instructions for writing tests for new code.
+
+We are using `pytest` to run tests on Firecrown.
+Before running tests, one must first build the code.
+In addition, the environment variable `PYTHONPATH` must be correctly set to run the tests.
+Please see the instructions, above, for this setup.
+
+### Use of `pylint`
+
+We are using `pylint` to check for a variety of possible problems.
+Firecrown is not currently "clean" of all `pylint` issues, so we are not yet using `pylint` in the CI testing.
+
+We are actively working on getting a "clean" report from `pylint`.
+When this is achieved, we plan to activate `pylint` in the CI checking.
+This will require that all new code pass `pylint`'s checks.
+
+### Code formatting
+
+We are using the command-line tool `black` to auto-format Python code in Firecrown.
+Please make sure to run black on your code before creating any commits.
+
+
+## Contact
+
+If you have comments, questions, or feedback, please [open an issue](https://github.com/LSSTDESC/firecrown/issues).
 
-See the examples in
-the [examples folder](https://github.com/LSSTDESC/firecrown/examples)
-for more details.
+You can also discuss Firecrown on the [#desc-firecrown](https://lsstc.slack.com/app_redirect?channel=desc-firecrown) LSSTC DESC Slack channel.
```

Comment:
```suggestion
You can also discuss Firecrown on the [#desc-firecrown](https://lsstc.slack.com/app_redirect?channel=desc-firecrown) LSSTC Slack channel.
```
(because the Slack workspace is from LSSTC, not from DESC - we are one of many groups that use it)